localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -190,19 +200,15 @@ def poll_events_from_shard(self, shard_id: str, shard_iterator: str):
                     LOG.warning(
                         "AUTOMATIC_BISECT upon partial batch item failure is not yet implemented. Retrying the entire batch."
                     )
-
+                error_payload = ex.partial_failure_payload
                 # let entire batch fail (ideally raise BatchFailureError)
+            except (SenderError, PartialFailureSenderError, BatchFailureError, Exception) as ex:
+                if isinstance(ex, (SenderError, PartialFailureSenderError, BatchFailureError)):
+                    error_payload = ex.error
+
+                # FIXME partner_resource_arn is not defined in ESM
                 LOG.debug(
-                    f"Attempt {attempts} failed while processing {self.partner_resource_arn} with events: {events}"
-                )
-                attempts += 1
-                # Retry polling until the record expires at the source
-                if self.stream_parameters.get("MaximumRetryAttempts", -1) == -1:
-                    # TODO: handle iterator expired scenario
-                    return
-            except Exception:
-                LOG.warning(
-                    f"Attempt {attempts} failed unexpectedly while processing {self.partner_resource_arn} with events: {events}"
+                    f"Attempt {attempts} failed while processing {self.partner_resource_arn or self.source_arn} with events: {events}"
```

Comment:
Just chiming in to say that when using Python logger, it should be preferred to use the logger string interpolation instead of f-strings, because the interpolation would happen only if the logging level is set, and not always like it would be the case with f-string.

See this Ruff rule: https://docs.astral.sh/ruff/rules/logging-f-string/